api/health/service_test.go

```diff
@@ -112,3 +113,83 @@ func TestServiceResponses(t *testing.T) {
 		require.True(reply.Healthy)
 	}
 }
+
+func TestServiceTagResponse(t *testing.T) {
+	require := require.New(t)
+
+	check := CheckerFunc(func(context.Context) (interface{}, error) {
+		return "", nil
+	})
+
+	subnetID1 := ids.GenerateTestID()
+	subnetID2 := ids.GenerateTestID()
+
+	h, err := New(logging.NoLog{}, prometheus.NewRegistry())
+	require.NoError(err)
+	err = h.RegisterHealthCheck("check1", check)
+	require.NoError(err)
+	err = h.RegisterHealthCheck("check2", check, subnetID1.String())
+	require.NoError(err)
+	err = h.RegisterHealthCheck("check3", check, subnetID2.String())
+	require.NoError(err)
+	err = h.RegisterHealthCheck("check4", check, subnetID1.String(), subnetID2.String())
+	require.NoError(err)
+
+	s := &Service{
+		log:    logging.NoLog{},
+		health: h,
+	}
+
+	// default checks
+	{
+		reply := APIReply{}
+		err = s.Health(nil, nil, &reply)
+		require.NoError(err)
+		require.Len(reply.Checks, 4)
+		require.Contains(reply.Checks, "check1")
+		require.Contains(reply.Checks, "check2")
+		require.Contains(reply.Checks, "check3")
+		require.Contains(reply.Checks, "check4")
+		require.Equal(notYetRunResult, reply.Checks["check1"])
+		require.False(reply.Healthy)
+
+		err = s.Health(nil, &HealthArgs{SubnetIDs: []ids.ID{subnetID1}}, &reply)
+		require.NoError(err)
+		require.Len(reply.Checks, 2)
+		require.Contains(reply.Checks, "check2")
+		require.Contains(reply.Checks, "check4")
+		require.Equal(notYetRunResult, reply.Checks["check2"])
+		require.False(reply.Healthy)
+	}
+
+	h.Start(context.Background(), checkFreq)
+	defer h.Stop()
+
+	awaitHealthy(h, true)
+
+	{
+		reply := APIReply{}
+		err = s.Health(nil, &HealthArgs{SubnetIDs: []ids.ID{subnetID1}}, &reply)
+		require.NoError(err)
+		require.Len(reply.Checks, 2)
+		require.Contains(reply.Checks, "check2")
+		require.Contains(reply.Checks, "check4")
+		require.True(reply.Healthy)
+	}
+
+	// now we'll add a new failing check
```

Comment:
Same comment as above -- doesn't `check` return healthy if it runs?

Reply:
changed it